core_lib/src/managers/layermanager.cpp

```diff
@@ -173,6 +175,34 @@ QString LayerManager::nameSuggestLayer(const QString& name)
     return newName;
 }
 
+void LayerManager::prepareRepositionSelectedFrames(int frame)
+{
+    if (editor()->select()->somethingSelected()) { return; }
+
+    if (currentLayer()->type() == Layer::BITMAP)
+    {
+        editor()->scrubTo(frame);
+        LayerBitmap* layer = static_cast<LayerBitmap*>(currentLayer());
+        QRect reposRect = layer->getFrameBounds(frame);
+        editor()->select()->setSelection(reposRect);
+    }
+}
+
+void LayerManager::repositionFrame(QPoint transform, int frame)
+{
+    if (currentLayer()->type() == Layer::BITMAP)
+    {
+        editor()->scrubTo(frame);
+        LayerBitmap* layer = static_cast<LayerBitmap*>(currentLayer());
+        QRect reposRect = layer->getFrameBounds(frame);
+        editor()->select()->setSelection(reposRect);
+        QPoint point = reposRect.topLeft();
+        point += transform;
+        layer->repositionFrame(point, frame);
+        editor()->backup(layer->id(), frame, tr("Reposition frame"));
+    }
+}
+
```

Comment:
This doesn't belong here, it has nothing to do with layer management. Should probably be placed in Editor for now.. but maybe another manager is needed.. a CanvasContentManager maybe.. not sure right now...

Reply:
I could move it to Editor, but what about Scribblearea? It is pretty much a CanvasContentManager, as it is now...